### PR TITLE
Rhcloud2463v2

### PR DESCRIPTION
**Pull request by yashnalla**
searching in host tags refinements

### CODE REVIEW

**Comment by yashnalla** [comment]
https://github.com/RedHatInsights/insights-host-inventory/pull/554

**Comment by Glutexo** [comment]
Can we close #585 now?

**Comment by yashnalla** [comment]
Yea I went ahead and closed it

File: api/host.py [hunk 1]
```diff
@@ -352,7 +352,7 @@ def get_host_tags(host_id_list, page=1, per_page=100, order_by=None, order_how=N
     return _build_paginated_host_tags_response(query.total, page, per_page, tags)
 
 
-def _build_serialized_tags(host_list, search):
+def _build_serialized_tags(host_list, search=None):
```
Comment (Glutexo) [inline api/host.py]: Is the default necessary? The function is called from [one single place](https://github.com/RedHatInsights/insights-host-inventory/blob/f94c8f8cb78cfee061168bbf5482edf875d47b94/api/host.py#L350) and the argument is always passed in.

File: test_unit.py [hunk 2]
```diff
@@ -495,10 +493,15 @@ def test_search_matches_tag_key(self):
         self._base_structured_to_filtered_test(structured_tags, expected_filtered_tags, "key1")
 
     def test_complex_filter(self):
```
Comment (Glutexo) [inline test_unit.py]: I still don’t know, why is this test called “complex”. 😄 But whatever.

File: test_unit.py [hunk 3]
```diff
@@ -495,10 +493,15 @@ def test_search_matches_tag_key(self):
...
-        structured_tags = [Tag("NS1", "key1", "val"), Tag(None), Tag("NS2", "key2"), Tag("NS3", "key3", "value3")]
-        expected_filtered_tags = [Tag("NS1", "key1", "val"), Tag("NS3", "key3", "value3")]
+        structured_tags = [
+            Tag("NS1erTys", "kindLey1", "v!2#eyal"),
+            Tag(None),
+            Tag("yeNS2", "ke56ty2"),
+            Tag("NYES12E3", "keingRty3", "value3uiley"),
+        ]
+        expected_filtered_tags = [Tag("NS1erTys", "kindLey1", "v!2#eyal"), Tag("NYES12E3", "keingRty3", "value3uiley")]
```
Comment (Glutexo) [inline test_unit.py]: Okay, let’s consider this nice _v!2#eyal_ string complex. 😅

**Comment by Glutexo** [comment]
Now I see that the commented out pieces of code are those that would make the [filter](https://github.com/RedHatInsights/insights-host-inventory/blob/f94c8f8cb78cfee061168bbf5482edf875d47b94/app/utils.py#L343) function possible to use with Python’s native filtering. Are you going to use those pieces or throw them out?

File: api/host.py (hunk 1 above)
Comment (yashnalla) [reply api/host.py]: that was an artifact from earlier I will go ahead an remove it

**Comment by Glutexo** [comment]
Overridden by #613. Still I’d like to see what was failing here for the sake of gaining experience and sharing knowledge.

**Pull request closed without merging**